src/Hosting/Hosting/src/WebHostBuilderExtensions.cs

```diff
@@ -22,6 +22,17 @@ public static class WebHostBuilderExtensions
         /// <param name="configureApp">The delegate that configures the <see cref="IApplicationBuilder"/>.</param>
         /// <returns>The <see cref="IWebHostBuilder"/>.</returns>
         public static IWebHostBuilder Configure(this IWebHostBuilder hostBuilder, Action<IApplicationBuilder> configureApp)
+        {
+            return hostBuilder.Configure((_, app) => configureApp(app));
```

Comment:
I see the problem. This added indirection breaks the following line in Configure:
https://github.com/aspnet/AspNetCore/blob/c3f361fc5395d3ef50b876a7f1850695371e9604/src/Hosting/Hosting/src/WebHostBuilderExtensions.cs#L42
The new declaring type for configureApp is WebHostBuilderExtensions, not Program.

This may be best addressed by adding a third private overload of Configure with most of the functionality but that takes startupAssemblyName and having both of these call into it.

Reply:
Thanks for the hint. Didn't spot that. Fixed now